server/src/main/java/io/crate/execution/engine/fetch/FetchCollector.java

```diff
@@ -67,27 +71,59 @@
 
     }
 
-    private void setNextDocId(LeafReaderContext readerContext, int doc) throws IOException {
+    private void setNextDocId(ReaderContext readerContext, int doc) throws IOException {
         for (LuceneCollectorExpression<?> e : collectorExpressions) {
             e.setNextReader(readerContext);
             e.setNextDocId(doc);
         }
     }
 
     public StreamBucket collect(IntContainer docIds) {
+        if (docIds.size() >= 10) {
```

Comment:
I experimented around with smaller values as size theshold since the amount of doc ids can be less due to parallel fetch execution in comparison to elasticsearch. However,  the performance goes worse with smaller batches.